helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -56,6 +57,24 @@
 import org.slf4j.LoggerFactory;
 
 public class ZkBaseDataAccessor<T> implements BaseDataAccessor<T> {
+
+  // Designates which mode ZkBaseDataAccessor should be created in. If not specified, it will be
+  // created on SHARED mode.
+  enum ZkClientType {
```

Comment:
I like this type. But I would prefer if we can put it in the HelixZkClientFactory layer. There are multiple options, either we add a method in a default factory implementation so it will generate both types of the ZkClient accordingly. Or we have util for that purpose to minimize the changes.

Reply:
The naming for this was deliberate: Moving forward, we won't necessarily be using HelixZkClientFactory. There will be other ZkClient APIs that are multiple ZK-aware that will serve as the ZkClient layer for this accessor. Does this make sense?